db/db.go

```diff
@@ -101,7 +101,15 @@ func (m *memKVStore) Delete(namespace string, key []byte) error {
 
 // Commit commits a batch
 func (m *memKVStore) Commit(b KVStoreBatch) (e error) {
+	succeed := false
 	b.Lock()
+	defer func() {
```

Comment:
i think the entire block can be simplified to defer b.Unlock() here, no need to add succeed, see comment below

Reply:
and yes, this way is better than the current Unlock() + ClearAndUnlock()